src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -84,6 +84,16 @@ export class TerminalService implements ITerminalService {
 		return this._terminalGroupService.instances.concat(this._terminalEditorService.instances);
 	}
 
+	/**
+	 * The task system is created after reconnection has occurred,
+	 * so it will later ask for the terminals that should be reconnected
+	 * to tasks
+	 */
+	private _taskReconnectedTerminals: ITerminalInstance[] = [];
```

Comment:
to do: make this generic. the event didn't work so well for tasks and might not for other owners either because of timing

Reply:
maybe should be a map from owner to terminals

Reply:
and `reconnectionOwner` can be the string key of the map